docs/mediasoup-server-protocol.md

```diff
@@ -7,234 +7,293 @@ This document provides detailed explanations of the request methods available in
-   ```
+
+- **Purpose**: Sets the priority of a media consumer.
+- **Usage**: Useful in bandwidth-limited scenarios to prioritize certain streams.
+
+**Example**:
+
+```javascript
+await socket.request('setConsumerPriority', { consumerId: consumer.id, priority: 1 });
+```
 
 ### 12. **requestConsumerKeyFrame**
-   - **Purpose**: Requests a keyframe from the producer.
-   - **Usage**: Refreshes the video stream for better quality.
-   
-   **Example**:
-   ```javascript
-   await socket.request('requestConsumerKeyFrame', { consumerId: consumer.id });
-   ```
+
+- **Purpose**: Requests a keyframe from the producer.
+- **Usage**: Refreshes the video stream for better quality.
+
+**Example**:
+
+```javascript
+await socket.request('requestConsumerKeyFrame', { consumerId: consumer.id });
+```
 
 ### 13. **produceData**
-   - **Purpose**: Produces data (non-media) via the WebRTC data channel.
-   - **Usage**: Used for sending arbitrary data, like chat messages.
-   
-   **Example**:
-   ```javascript
-   const dataProducer = await transport.produceData({ label: 'chat' });
-   ```
+
+- **Purpose**: Produces data (non-media) via the WebRTC data channel.
+- **Usage**: Used for sending arbitrary data, like chat messages.
+
+**Example**:
+
+```javascript
+const dataProducer = await transport.produceData({ label: 'chat' });
+```
 
 ### 14. **getTransportStats / getProducerStats / getConsumerStats**
-   - **Purpose**: Retrieves statistics for transports, producers, and consumers.
-   - **Usage**: Useful for monitoring WebRTC session performance.
-   
-   **Example** (Transport Stats):
-   ```javascript
-   const stats = await socket.request('getTransportStats', { transportId: transport.id });
-   ```
-
-   **Example** (Producer Stats):
-   ```javascript
-   const producerStats = await socket.request('getProducerStats', { producerId: producer.id });
-   ```
-
-   **Example** (Consumer Stats):
-   ```javascript
-   const consumerStats = await socket.request('getConsumerStats', { consumerId: consumer.id });
-   ```
+
+- **Purpose**: Retrieves statistics for transports, producers, and consumers.
+- **Usage**: Useful for monitoring WebRTC session performance.
+
+**Example** (Transport Stats):
+
+```javascript
+const stats = await socket.request('getTransportStats', { transportId: transport.id });
+```
+
+**Example** (Producer Stats):
+
+```javascript
+const producerStats = await socket.request('getProducerStats', { producerId: producer.id });
+```
+
+**Example** (Consumer Stats):
+
+```javascript
+const consumerStats = await socket.request('getConsumerStats', { consumerId: consumer.id });
+```
+
+### 15. **leaveCall**
+
+- **Purpose**: Allows a peer to notify the server when leaving a call, ensuring resource cleanup and notifying other peers in the room.
+- **Usage**: Sends a signal to the server to terminate the peer's session and triggers notifications to other participants.
+
+**Example**:
+
+```javascript
+await socket.request('leaveCall');
```

Comment:
```suggestion
await socket.request('leaveRoom');
```

This is mostly because within the code we always talk about 'rooms' rather than 'calls'